astronomer/airflow/version_check/plugin.py

```diff
@@ -78,6 +90,80 @@ def start_update_thread(cls):
         AstronomerVersionCheck.ensure_singleton()
         CheckThread().start()
 
+    @classmethod
+    def get_migration_columns(cls):
+        """Return the columns to be added during migration."""
+        from .models import AstronomerAvailableVersion
+
+        return {
+            AstronomerAvailableVersion.__tablename__: [
+                Column('end_of_support', UtcDateTime(timezone=True), nullable=True),
+                Column('eos_dismissed_until', UtcDateTime(timezone=True), nullable=True),
+            ],
+        }
+
+    @classmethod
+    def migrate_db_tables_and_upsert(cls):
+        """Apply migrations to the DB tables and upsert data."""
+        with create_session() as session:
+            cls.create_columns_for_migration(session)
+            cls.insert_initial_data(session)
```

Comment:
After checking, I found that another process may have a lock since it is always returning `None`. Now I am trying to identify which process has the lock.

Reply:
I think the issue here is this line when we set `AIRFLOW__ASTRONOMER_UPDATE_CHECK_INTERVAL=1`, it checks after a day based on: https://github.com/astronomer/astronomer-airflow-version-check/blob/a4bd891d0bf51f22850e2cc3c017f315ef604470/astronomer/airflow/version_check/update_checks.py#L181

When I changed these lines to 
```
                if not lock:
                    next_check = self.check_interval
                    how_long = (next_check - utcnow()).total_seconds()
                    self.log.info("Next check not due until %s (%s seconds away)", next_check, how_long)
                    return UpdateResult.NOT_DUE, how_long
```

It works fine in astro cloud as well. Deployment is here: https://cloud.astronomer-stage.io/cll0nk0c3003u01kd092pghag/deployments/clyed2zxu02tn01hxzjt1ybc3/logs

Reply:
cc: When the `AIRFLOW__ASTRONOMER_UPDATE_CHECK_INTERVAL` is set, its initial value is taken into consideration for when to check next in the future. In the current code, it will always be 1 day after.

Reply:
You have a typo in your env var name: `AIRFLOW__ASTRONOMER__UPDATE_CHECK_INTERVAL=1`

Reply:
As per discussion with @jedcunningham we should explore resetting last checked to `None` if we migrated any fields.

Reply:
Changed it @jedcunningham https://github.com/astronomer/astronomer-airflow-version-check/pull/72/commits/08a06790d79df2f799d7e78f28e40c59368fadb1